### PR TITLE
Carli/2162 fix text annotation box size

### PR DESCRIPTION
**Pull request by crocka**
**Description**
The factor ``frame.zoomLevel / AppStore.Instance.imageRatio`` is added to the text annotation textbox so that the size of the textbox won't change with zoom.

**Checklist**

For linked issues (if there are):
- [x] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] e2e test passing / corresponding fix added
- [x] changelog updated / ~no changelog update needed~
- [x] ~~protobuf updated to the latest dev commit~~ / no protobuf update needed
- [x] `BackendService` unchanged / ~~`BackendService` changed and corresponding ICD test fix added~~

### CODE REVIEW

**Review by kswang1029 — APPROVED** [review]
this is looking good now. no regression from e2e. 👍

File: src/components/Dialogs/RegionDialog/CompassRulerRegionForm/CompassRulerRegionForm.tsx [hunk 1]
```diff
@@ -7,15 +7,25 @@ import {observer} from "mobx-react";
...
 import {AppStore} from "stores";
-import {CompassAnnotationStore, CoordinateMode, RegionStore} from "stores/Frame";
+import {CompassAnnotationStore, CoordinateMode, FrameStore, RegionStore} from "stores/Frame";
 import {getFormattedWCSPoint, getPixelValueFromWCS, isWCSStringFormatValid} from "utilities";
 
+const KEYCODE_ENTER = 13;
+
 @observer
-export class CompassRulerRegionForm extends React.Component<{region: RegionStore; wcsInfo: AST.FrameSet}> {
+export class CompassRulerRegionForm extends React.Component<{region: RegionStore; frame: FrameStore; wcsInfo: AST.FrameSet}> {
```
Comment (YuHsuan-Hwang) [inline src/components/Dialogs/RegionDialog/CompassRulerRegionForm/CompassRulerRegionForm.tsx]: Remove `props.frame` if it's not used.

File: src/components/Dialogs/RegionDialog/RectangularRegionForm/RectangularRegionForm.tsx [hunk 2]
```diff
@@ -123,52 +123,57 @@ export class RectangularRegionForm extends React.Component<{region: RegionStore;
         return false;
     };
 
-    private handleWidthChange = (value: number): boolean => {
+    private handleWidthChange = (value: number, fixedScreenSize: boolean = false): boolean => {
         const existingValue = this.props.region.size.x;
+        const constant = fixedScreenSize ? AppStore.Instance.imageRatio / this.props.frame.zoomLevel : 1;
```
Comment (YuHsuan-Hwang) [inline src/components/Dialogs/RegionDialog/RectangularRegionForm/RectangularRegionForm.tsx]: Perhaps rename to `scale` is more informative? Same for other functions.

File: src/components/Dialogs/RegionDialog/RectangularRegionForm/RectangularRegionForm.tsx [hunk 3]
```diff
@@ -334,24 +343,24 @@ export class RectangularRegionForm extends React.Component<{region: RegionStore;
         // bottom left
         const bottomLeftPoint = this.bottomLeftPoint;
         const bottomLeftWCSPoint = this.bottomLeftWCS;
-        const bottomLeftInputX = (
+        const bottomLeftInputX = (isTextAnnotation: boolean = false) => (
```
Comment (YuHsuan-Hwang) [inline src/components/Dialogs/RegionDialog/RectangularRegionForm/RectangularRegionForm.tsx]: We could directly use `isTextAnnotation` without adding an input parameter. Same for other functions.

**Review by YuHsuan-Hwang — CHANGES_REQUESTED** [review]
The changes look good. I only have very minor comments.

File: src/components/Dialogs/RegionDialog/RectangularRegionForm/RectangularRegionForm.tsx (hunk 3 above)
Comment (YuHsuan-Hwang) [reply src/components/Dialogs/RegionDialog/RectangularRegionForm/RectangularRegionForm.tsx]: @crocka Could we also change functions `{sizeWidthInput()}` back to elements `{sizeWidthInput}`?
Other than this, it's ready for merge. Please also update the change log, thanks.

**Review by YuHsuan-Hwang — APPROVED** [review]

**Pull request merged**